File: android/src/main/kotlin/com/arthurmonteiroo/perimeterx_flutter_plugin/PerimeterxFlutterPlugin.kt
package com.arthurmonteiroo.perimeterx_flutter_plugin

import com.arthurmonteiroo.perimeterx_flutter_android.messages.PerimeterXHostApi
import io.flutter.embedding.engine.plugins.FlutterPlugin

/** PerimeterxFlutterPlugin */
class PerimeterxFlutterPlugin: FlutterPlugin {
  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    PerimeterXHostApi.setUp(flutterPluginBinding.binaryMessenger, PerimeterXHostApiImpl())
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: ios/Classes/PerimeterxFlutterPlugin.swift
import Flutter
import UIKit

public class PerimeterxFlutterPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
      PerimeterXHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: PerimeterXHostApiImpl())
  }
}
